sdk/keyvault/keyvault-keys/src/keyVaultCryptographyClient.ts

```diff
@@ -0,0 +1,690 @@
+          span.end();
+          throw e;
+        }
+      }
+    }
+
+    const createHash: LocalCryptographyOperationFunction = localSupportedAlgorithms[localAlgorithm]
+      ?.operations.createHash as LocalCryptographyOperationFunction;
+    const digest = await createHash("", Buffer.from(data));
+
+    // Default to the service
+
+    let result;
+    try {
+      result = await this.client.verify(
+        this.vaultUrl,
+        this.name,
+        this.version,
+        algorithm,
+        digest,
+        signature,
+        setParentSpan(span, requestOptions)
+      );
+    } finally {
+      span.end();
+    }
+
+    return { result: result.value!, keyID: this.getKeyID() };
+  }
+
+  /**
+   * @internal
+   * @hidden
+   * Attempts to retrieve the ID of the key.
+   */
+  private getKeyID(): string | undefined {
+    let kid;
+    if (typeof this.key !== "string") {
+      kid = this.key.kid;
+    } else {
+      kid = this.key;
+    }
+
+    return kid;
+  }
+
+  /**
+   * The base URL to the vault.
+   */
+  public readonly vaultUrl: string;
+
+  /**
+   * The ID of the key used to perform cryptographic operations for the client.
+   */
+  public get keyId(): string | undefined {
+    return this.getKeyID();
+  }
+
+  /**
+   * @internal
+   * @hidden
+   * A reference to the auto-generated KeyVault HTTP client.
+   */
+  private readonly client: KeyVaultClient;
+
+  /**
+   * Other relevant information of the Key Vault Key that will be used for cryptographic operations.
+   */
+  private keyBundle: KeyBundle | undefined;
+
+  /**
+   * A reference to the key used for the cryptographic operations.
+   * Based on what was provided to the CryptographyClient constructor, it can be either a string with the URL of a Key Vault Key, or an already parsed {@link JsonWebKey}.
+   */
+  private key: string | JsonWebKey;
+
+  /**
+   * Name of the key the client represents
+   */
+  private name: string;
+
+  /**
+   * Version of the key the client represents
+   */
+  private version: string;
+
+  /**
+   * Tries to load the full Key Vault Key and then creates the Local Cryptography Client
+   */
+  private _localCryptographyClientPromise?: Promise<LocalCryptographyClient>;
+
+  /**
+   * Returns a promise that resolves once the Local Cryptography Client is available.
+   */
+  private getLocalCryptographyClient(): Promise<LocalCryptographyClient> | undefined {
+    if (!this._localCryptographyClientPromise) {
+      this._localCryptographyClientPromise = this.configureLocalCryptographyClient();
+    }
+    return this._localCryptographyClientPromise;
+  }
+
+  private async configureLocalCryptographyClient(): Promise<LocalCryptographyClient> {
+    this.key = await this.getKey();
+    return new LocalCryptographyClient(this.key as JsonWebKey);
+  }
+
+  constructor(
+    key: string | KeyVaultKey,
+    credential: TokenCredential,
+    pipelineOptions: CryptographyClientOptions = {}
```

Comment:
I noticed this with the Search client when trying to help a customer: it seems you have no way for the callers to pass in additional policies. Is that expected? I know .NET and Java allow it.

Reply:
We chatted on teams - I will follow up on this separately. Thanks for pointing this out @heaths  !